src/display/api.js

```diff
@@ -281,8 +293,19 @@ function getDocument(src, pdfDataRangeTransport,
       if (task.destroyed) {
         throw new Error('Loading aborted');
       }
+
+      let networkStream;
+      if (rangeTransport) {
+        networkStream = new PDFDataTransportStream(params, rangeTransport);
+      } else {
+        networkStream =
```

Comment:
> Also tried fixing viewer progress bar, can you please check if it works for you?

Yes, it appears that the progress bar is now updating correctly in all configurations that I've tested (with/without `disableRange`, `disableStream`, `disableAutoFetch`, and combinations of them).